File: src/main/java/io/github/przbetkier/tuscan/domain/profiles/LatestProfileRepository.java
package io.github.przbetkier.tuscan.domain.profiles;

import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;

@Repository
interface LatestProfileRepository extends ReactiveMongoRepository<LatestProfile, String> {

    Flux<LatestProfile> findTop6ByOrderByCreatedOnDesc();

}

File: src/main/java/pro/tuscan/domain/profiles/LatestProfileRepository.kt
package pro.tuscan.domain.profiles

import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

interface LatestProfileRepository: ReactiveMongoRepository<LatestProfile, String> {
    fun findTop6ByOrderByCreatedOnDesc(): Flux<LatestProfile>
}
